### DISCUSSION TITLE
Streams - gzip-server with media file

### DISCUSSION THREAD

**Discussion by stuartambient** [PacktPublishing/Node.js-Design-Patterns-Third-Edition#30] [Q&A]
I was wondering if the headers would need to be changed zipping a media file (.mkv).  Turned out to be a non-issue, apologies!

**Comment by lmammino** [comment]
Did you end up using `Content-Encoding: gzip` ?

**Reply by stuartambient** [reply]
> 
> 
> Did you end up using `Content-Encoding: gzip` ?

I have not changed the encoding.  
My study method is to type out the code from the book and check the github code to make sure I copied correctly.  While my gzip-server code matches the code correctly from the repo, mine doesn't work. It makes it as far as the server showing "file request received" correctly and then never goes any further.  Using the same file with the repo code it works fine.  

Comparing versions side by side I've yet to find a typo in my code so still trying to figure out where the issue lies.
I'll update thread once I find the problem.  

**Update:** I discovered an escape character was needed in gzip-receive.js , `i.e. res.end('OK\\n')`
Had it throw an error and discover it.

It's all good now and compression works fine.  I might try and save the compressed file as well, if it's possible.   It would give  me an idea how files get compressed. 

Thank you!

**Reply by lmammino** [reply]
I think I might have misunderstood your original comment... I didn't realize you were trying to re-create a specific example from the book...

That new line character is an interesting one. I don't think it should be needed... Can you share your final working code, please?

**Reply by stuartambient** [reply]
Sorry it's take me a while to reply.   

All I'm really doing is recreating the code instead of running it from the repo code.  I find that in some cases though it just doesn't work. No errors though and the program seems to exit properly.  I'm running Node 14.3.0,  As an example,'to-file-stream.js', #11 in chapter 6.  I set up a folder, had the two files, plus the package.json, and installed mkdirp-promise (same version).  I had repo code and mine side by side checking for typos.  Only 1 write request made it through, the 'files' directory was created and the program exits. No files written.

I ran the repo code with no problem.  I then copy the repo code, replacing what I typed, still same behavior.  Finally I replace my folder with the #11 folder, run npm i, run the app and it works.  

I even had added the engine and engineStrict to my package.json with no effect.   This has got to be my problem but it's leaving me confused.

edit: I also tried it in linux with no change.
